test/unit/lib/TopSitesFeed.test.js

```diff
@@ -1088,35 +1088,60 @@ describe("Top Sites Feed", () => {
       feed.store.state.Prefs.values[NO_DEFAULT_SEARCH_TILE_PREF] = true;
     });
     afterEach(() => {
-      global.Services.search.defaultEngine = cachedDefaultSearch;
+      global.Services.search.currentEngine = cachedDefaultSearch;
+    });
+    it("should filter out alexa top 5 search from the default sites", async () => {
+      const TOP_5_TEST = [
+        "google.com",
+        "search.yahoo.com",
+        "yahoo.com",
+        "bing.com",
+        "ask.com",
+        "duckduckgo.com"
+      ];
+      links = [{url: "amazon.com"}, ...TOP_5_TEST.map(url => ({url}))];
+      const urlsReturned = (await feed.getLinksWithDefaults()).map(link => link.url);
+      assert.include(urlsReturned, "amazon.com");
+      TOP_5_TEST.forEach(url => assert.notInclude(urlsReturned, url));
     });
-    it("should not filter out google from the query results if the experiment pref is off", async () => {
-      links = [{url: "google.com"}, {url: "foo.com"}];
+    it("should not filter out alexa, default search from the query results if the experiment pref is off", async () => {
+      links = [{url: "google.com"}, {url: "foo.com"}, {url: "duckduckgo"}];
       feed.store.state.Prefs.values[NO_DEFAULT_SEARCH_TILE_PREF] = false;
       const urlsReturned = (await feed.getLinksWithDefaults()).map(link => link.url);
       assert.include(urlsReturned, "google.com");
     });
-    it("should filter out google from the default sites if it matches the current default search", async () => {
+    it("should filter out the current default search from the default sites", async () => {
+      feed._currentSearchHostname = "amazon";
       feed.onAction({type: at.PREFS_INITIAL_VALUES, data: {"default.sites": "google.com,amazon.com"}});
       links = [{url: "foo.com"}];
       const urlsReturned = (await feed.getLinksWithDefaults()).map(link => link.url);
-      assert.include(urlsReturned, "amazon.com");
-      assert.notInclude(urlsReturned, "google.com");
+      assert.notInclude(urlsReturned, "amazon.com");
     });
-    it("should not filter out google from pinned sites even if it matches the current default search", async () => {
+    it("should not filter out current default search from pinned sites even if it matches the current default search", async () => {
       links = [{url: "foo.com"}];
       fakeNewTabUtils.pinnedLinks.links = [{url: "google.com"}];
       const urlsReturned = (await feed.getLinksWithDefaults()).map(link => link.url);
       assert.include(urlsReturned, "google.com");
     });
-    it("should call refresh when the the default search engine has been set", () => {
+    it("should set ._currentSearchHostname to the current engine hostname on init", async () => {
+      global.Services.search.currentEngine = {identifier: "ddg", searchForm: "duckduckgo.com"};
+      sandbox.stub(feed, "refresh");
+
+      await feed.init();
+
+      assert.equal(feed._currentSearchHostname, "duckduckgo");
```

Comment:
Testing this behavior with a default search that is also normally filtered out by `SEARCH_FILTERS` might not be ideal, but I suppose this unit test is only checking that `_currentSearchHostname` gets set. Not that `isExperimentOnAndLinkFilteredSearch` correctly filters the current search.